src/python/zquantum/core/wip/circuits/_gates_test.py

```diff
@@ -175,3 +193,39 @@ def test_binding_parameters_in_control_gate_is_propagated_to_wrapped_gate(
         controlled_gate = gate.controlled(2)
         symbols_map = {sympy.Symbol("theta"): 0.5, sympy.Symbol("x"): 3}
         assert controlled_gate.bind(symbols_map) == gate.bind(symbols_map).controlled(2)
+
+    def test_constructing_controlled_gate_and_replacing_parameters_commute(
+        self, gate
+    ):
+        controlled_gate = gate.controlled(2)
+        new_params = tuple(3 * param for param in controlled_gate.params)
+
+        assert (
+            controlled_gate.replace_params(new_params) ==
+            gate.replace_params(new_params).controlled(2)
+        )
+
+
+@pytest.mark.parametrize("gate", GATES_REPRESENTATIVES)
+class TestGateOperation:
+
+    def test_binding_parameters_constructs_operation_of_gate_with_bound_parameters(
+        self, gate
+    ):
+        op = GateOperation(gate, tuple(range(gate.num_qubits)))
+        symbols_map = {sympy.Symbol("phi"): 0.5, sympy.Symbol("y"): 1.1}
+        assert (
+            op.bind(symbols_map) ==
+            GateOperation(gate.bind(symbols_map), tuple(range(gate.num_qubits)))
+        )
```

Comment:
This assertion duplicates implementation of the code it's supposed to verify. WDYT about changing this test's assertion to a one that would reflect the test case name?
```suggestion
        assert set(op.bind(symbols_map).gate.params) == set(symbols_map.values())
```